### ISSUE TITLE
insert ... return $current.link1:{field1} syntax returns null in develop branch

### ISSUE THREAD

**Issue by freeart** [orientechnologies/orientdb#8543]
## Expected behavior  
```
{id: "xxx", name: "xxx"}  
```

## Actual behavior 
``` 
{id: null, name: null}  
```
```Update``` and ```Select``` works fine, but ```Insert```doesn't return values of fields of LINK type.

## Steps to reproduce
```sql  
insert into `Table`
set code = #27:49 
RETURN $current.code:{id,name}
```

**Comment by luigidellaquila** [comment]
Hi @freeart 

I just tried the following on develop branch and it seems to work fine

```
create class Table;
create class Code;

insert into Code set id = 10, name= 'foo';

insert into `Table` set code = #242:0 RETURN $current.code:{id,name};
```

Am I missing something?

Thanks

Luigi

**Comment by freeart** [comment]
I found the difference between your query and my query. I think it is a bug 
```sql
create class TCode;
CREATE PROPERTY `TCode`.`id` STRING
ALTER PROPERTY `TCode`.`id` mandatory true
ALTER PROPERTY `TCode`.`id` readonly false
ALTER PROPERTY `TCode`.`id` notNull true
CREATE PROPERTY `TCode`.`name` STRING
ALTER PROPERTY `TCode`.`name` mandatory true
ALTER PROPERTY `TCode`.`name` readonly false
ALTER PROPERTY `TCode`.`name` notNull true

create class TTable;
CREATE PROPERTY `TTable`.`code` LINK  `TCode`
ALTER PROPERTY `TTable`.`code` linkedClass "TCode"
ALTER PROPERTY `TTable`.`code` mandatory true
ALTER PROPERTY `TTable`.`code` readonly false
ALTER PROPERTY `TTable`.`code` notNull true
CREATE PROPERTY `TTable`.`name` STRING
ALTER PROPERTY `TTable`.`name` mandatory true
ALTER PROPERTY `TTable`.`name` readonly false
ALTER PROPERTY `TTable`.`name` notNull true

insert into TCode set id = 10, name= 'foo';
```

So, this sql 
```sql
insert into `TTable` 
set 
	code = (SELECT @rid from TCode WHERE id = 10),
    name = "test"
RETURN 
	$current.code:{id,name} as code,
    $current.name as name;
```
returns ```{"id":null,"name":null} | test```
but, 
```sql
insert into `TTable` 
set 
	code = (SELECT from TCode WHERE id = 10),
    name = "test"
RETURN 
	$current.code:{id,name} as code,
    $current.name as name;
```
returns ```{"id":10,"name":"foo"} | test```

**Comment by luigidellaquila** [comment]
OK, it makes sense, `SELECT @rid from TCode` returns a projection with a single column, not a full document with its attributes, so you have to refer to that column to 

Just to clarify, both the following will work fine:

```
insert into `TTable` 
set 
	code = (SELECT @rid from TCode WHERE id = 10),
    name = "test"
RETURN 
	$current.code.@rid:{id,name} as code,
    $current.name as name;
```

```
insert into `TTable` 
set 
	code = (SELECT @rid as foo from TCode WHERE id = 10),
    name = "test"
RETURN 
	$current.code.foo:{id,name} as code,
    $current.name as name;
```

Thanks

Luigi

**Comment by freeart** [comment]
Any news? I'm afraid to remove @rid in all cases because it can break other functionality.

**Comment by luigidellaquila** [comment]
Hi @freeart 

I'm pushing a fix now, it will be released with v 3.0.12

Thanks

Luigi
